api/post.go

```diff
@@ -728,6 +730,92 @@ func updateMentionCountAndForget(channelId, userId string) {
 	}()
 }
 
+func checkForOutOfChannelMentions(c *Context, post *model.Post, channel *model.Channel) {
+	// don't check for out of channel mentions in direct channels
+	if channel.Type == model.CHANNEL_DIRECT {
+		return
+	}
+
+	mentioned := getOutOfChannelMentions(post, channel.TeamId)
+	if len(mentioned) == 0 {
+		return
+	}
+
+	usernames := make([]string, len(mentioned))
+	for i, user := range mentioned {
+		usernames[i] = user.Username
+	}
+	sort.Strings(usernames)
+
+	var messageText string
```

Comment:
@enahum I didn't use go-i18n's support for pluralized strings in the post message because I didn't know how to make it work with a comma-separated list of values. Do you know of any better way to do this instead of manually adding commas between the usernames?